### PR TITLE
Ami remover

### PR DESCRIPTION
**Pull request by Itxaka**
Fixes https://github.com/rancher-sandbox/cOS-toolkit/issues/1190
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: .github/ami-cleaner.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  export AWS_DEFAULT_REGION=${region}
+  ami_number=$( aws ec2 describe-images --owners $AMI_OWNER | jq '.Images| length' )
+  snapshots_number=$( aws ec2 describe-snapshots --owner-id $AMI_OWNER | jq '.Snapshots| length' )
+  printf "\xE2\x86\x92 Found %s AMIs\n" "$ami_number"
+  printf "\xE2\x86\x92 Found %s Snapshots\n" "$snapshots_number"
+  if [[ $ami_number > $MAX_AMI_NUMBER ]]; then
+    printf "\xE2\x86\x92 AMI number is bigger than max number allowed, engaging cleanup.\n"
+    to_clean=$(expr $ami_number - $MAX_AMI_NUMBER)
+    printf "\xE2\x86\x92 Cleaning %s AMIs+Snapshots\n" "$to_clean"
+    to_clean_amis=$(aws ec2 describe-images --owners 053594193760 |jq -r ".Images | sort_by(.CreationDate) | .[:${to_clean}] | .[]| {ImageId: .ImageId, CreationDate: .CreationDate, SnapshotId: .BlockDeviceMappings[].Ebs.SnapshotId}")
```
Comment (davidcassany) [inline .github/ami-cleaner.sh:35]: Looks like this should consume the `AMI_OWNER` variable instead of the literal

**Review by davidcassany — APPROVED** [review]
Looks good to me, just a couple of comments regarding the AMI_OWNER variable usage

File: .github/ami-cleaner.sh (hunk 1 above)
Comment (Itxaka) [reply .github/ami-cleaner.sh:35]: Wooops, good catch

**Comment by Itxaka** [comment]
This is a real run I did to clean up and leave up to 20 images, starting with most recent

```
$ DO_CLEANUP=true .github/ami-cleaner.sh
------------------------------------------------------------------
💀 Cleaning mode activated
Maximun number of AMIs allowed: 20
Owner of the AMIs: 053594193760
Regions: eu-north-1 ap-south-1 eu-west-3 eu-west-2 eu-west-1 ap-northeast-3 ap-northeast-2 ap-northeast-1 sa-east-1 ca-central-1 ap-southeast-1 ap-southeast-2 eu-central-1 us-east-1 us-east-2 us-west-1 us-west-2
------------------------------------------------------------------
→ Checking AMIS on region eu-north-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ap-south-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region eu-west-3 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region eu-west-2 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region eu-west-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ap-northeast-3 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ap-northeast-2 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ap-northeast-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region sa-east-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ca-central-1 for owner 053594193760
→ Found 20 AMIs
→ Found 20 Snapshots
✅ Ami number is equal or lower to max number allowed, skipping.
→ Checking AMIS on region ap-southeast-1 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-0f1c9be53578dea26
❌ Removing AMI ami-024382f2c6aaf3557
❌ Removing Snapshot snap-09fc066a89295a998
❌ Removing Snapshot snap-0b4a52c77fd601a2f
→ Checking AMIS on region ap-southeast-2 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-0c9c7ecaf320eb96d
❌ Removing AMI ami-0e3350aac60049b6c
❌ Removing Snapshot snap-03bd89bc57a67d36b
❌ Removing Snapshot snap-0cb6f1ee9ec407ea0
→ Checking AMIS on region eu-central-1 for owner 053594193760
→ Found 39 AMIs
→ Found 41 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 19 AMIs+Snapshots
❌ Removing AMI ami-0e1ae019b01f4b166
❌ Removing AMI ami-0208e174dc82acfbe
❌ Removing AMI ami-0dc7e57e5d30c6c49
❌ Removing AMI ami-05bbf127137397e67
❌ Removing AMI ami-08deadec2b216379f
❌ Removing AMI ami-0efb5a37b41b1bacf
❌ Removing AMI ami-0f2f8619e8a001989
❌ Removing AMI ami-0b89f2a6d5ca052b1
❌ Removing AMI ami-0325d41f0647342e1
❌ Removing AMI ami-03183dd0d723a47e4
❌ Removing AMI ami-0b04d7ca45aff1475
❌ Removing AMI ami-04a8ea3c039541165
❌ Removing AMI ami-0484112796575e489
❌ Removing AMI ami-0a933292d58e1848e
❌ Removing AMI ami-04985b3b252d2d0a8
❌ Removing AMI ami-0c76610be55ec3157
❌ Removing AMI ami-0f553b8976d5dab56
❌ Removing AMI ami-059cef73a61b48074
❌ Removing AMI ami-0642cf809b2e8b2d7
❌ Removing Snapshot snap-04e7db1dcdc95f942
❌ Removing Snapshot snap-0103a8efa91dad91d
❌ Removing Snapshot snap-02640a4c9811ad20a
❌ Removing Snapshot snap-02e4d24f311ba1846
❌ Removing Snapshot snap-080cf201279d55c08
❌ Removing Snapshot snap-035f680c273e167f0
❌ Removing Snapshot snap-0a386235645074bfc
❌ Removing Snapshot snap-049fdbdc6b091fe17
❌ Removing Snapshot snap-06d19ab5724003948
❌ Removing Snapshot snap-052133c87cbaf9a00
❌ Removing Snapshot snap-0478156c81fa0c87c
❌ Removing Snapshot snap-0929e50fe158ad87e
❌ Removing Snapshot snap-02b557e4c20194e75
❌ Removing Snapshot snap-07e3ad4a16bbb4853
❌ Removing Snapshot snap-07398902b5d206ccf
❌ Removing Snapshot snap-06a1da2df8247dc36
❌ Removing Snapshot snap-0884e55e6f46a7408
❌ Removing Snapshot snap-06ac4e3e4f7ca8b32
❌ Removing Snapshot snap-0f52c38193b47d476
→ Checking AMIS on region us-east-1 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-0b176747372b8c69e
❌ Removing AMI ami-0e681afa950a1b20c
❌ Removing Snapshot snap-04d67c4e1d8fe71e3
❌ Removing Snapshot snap-04510d41976a30ae4
→ Checking AMIS on region us-east-2 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-06790ab439ef41499
❌ Removing AMI ami-06baade2070fafe55
❌ Removing Snapshot snap-0cf5bb01887564ca3
❌ Removing Snapshot snap-0eefc8bd73462fe34
→ Checking AMIS on region us-west-1 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-09d7702c296bddee5
❌ Removing AMI ami-071551e5383dd02b6
❌ Removing Snapshot snap-043907432628d1fa9
❌ Removing Snapshot snap-0661942de752bccff
→ Checking AMIS on region us-west-2 for owner 053594193760
→ Found 22 AMIs
→ Found 22 Snapshots
→ AMI number is bigger than max number allowed, engaging cleanup.
→ Cleaning 2 AMIs+Snapshots
❌ Removing AMI ami-0d46d7215bcd3e15c
❌ Removing AMI ami-02d5c602c7bd86d4b
❌ Removing Snapshot snap-04319ef49665b9542
❌ Removing Snapshot snap-0d16f515d56752f22
```

**Comment by Itxaka** [comment]
Also missing the volume cleaner, no idea how to link that thought

**Comment by Itxaka** [comment]
Added orphan snapshot.

Tested everything and seems to work as expected, would like some eyes on this thought in case something is missing or can be broken.

File: .github/ami-cleaner.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
```
Comment (mudler) [inline .github/ami-cleaner.sh:1]: mmm just a thought, maybe this should all go with `set -e` to avoid that any aws call/escaping thing goes bad

**Review by mudler — APPROVED** [review]
Beside small nit, it looks good

**Review by davidcassany — APPROVED** [review]
LGTM

File: .github/ami-cleaner.sh (hunk 2 above)
Comment (Itxaka) [reply .github/ami-cleaner.sh:1]: good point

**Comment by Itxaka** [comment]
note that this has the cleanup set as false for now. Will keep it autorunning a couple of days before enabling it for real

**Pull request merged**